projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx

```diff
@@ -443,6 +447,39 @@ export const RemoveLiquidity = ({ well, slippage, slippageSettingsClickHandler,
   );
 };
 
+export const RemoveLiquidity: React.FC<{ well: Well | undefined; loading: boolean } & BaseRemoveLiquidityProps> = (props) => {
+  if (!props.well || props.loading) {
+    return (
```

Comment:
Same as before, separate component would make sense here... (Will not comment on other similar items in the rest of the review)